tests/test_layer_parameters.py

```diff
@@ -0,0 +1,285 @@
+
+def get_crs_from_image_data(image_data):
+    crs_string = image_data.rio.crs.data['init']
+    crs = CRS.from_string(crs_string)
+    return crs
+
+
+def get_modified_resolution_data(class_instance, spatial_resolution):
+    class_instance.spatial_resolution = spatial_resolution
+    data = class_instance.get_data(BBOX)
+    return data
+
+def validate_layer_instance(obj):
+    is_valid = True
+    except_str = None
+
+    if not obj.__class__.__bases__[0] == Layer:
+        is_valid = False
+        except_str = "Specified object '%s' is not a valid Layer class instance." % obj
+    else:
+        cls = get_class_from_instance(obj)
+        cls_name = type(cls).__name__
+        if not hasattr(obj, 'spatial_resolution'):
+            is_valid = False
+            except_str = "Class '%s' does not have spatial_resolution property." % cls_name
+        elif not obj.spatial_resolution == cls.spatial_resolution:
+            is_valid = False
+            except_str = "Do not specify spatial_resolution property value for class '%s'." % cls_name
+        elif cls.spatial_resolution is None:
+            is_valid = False
+            except_str = "Signature of class %s must specify a non-null default value for spatial_resolution. Please correct." % cls_name
+
+    return is_valid, except_str
+
+def estimate_spatial_resolution(data):
+    y_cells = float(data['y'].size - 1)
+    y_min = data.coords['y'].values.min()
+    y_max = data.coords['y'].values.max()
+
+    crs = get_crs_from_image_data(data)
+    crs_unit = crs.axis_info[0].unit_name
+
+    if crs_unit == 'metre':
+        diff_distance = y_max - y_min
+    else:
+        raise Exception('Unhandled projection units: %s for projection: %s' % (crs_unit, crs.srs))
+
+    estimated_actual_resolution = round(diff_distance / y_cells)
+
+    return estimated_actual_resolution
+
+def get_populate_ratio(dataset):
+    raw_data_size = dataset.values.size
+    populated_raw_data = dataset.values[(~np.isnan(dataset.values)) & (dataset.values > 0)]
+    populated_data_raw_size = populated_raw_data.size
+    populated_raw_data_ratio = populated_data_raw_size/raw_data_size
+    return populated_raw_data_ratio
+
+def evaluate_raster_value(raw_data, downsized_data):
+    # Below values where determined through trial and error evaluation of results in QGIS
+    ratio_tolerance = 0.2
+    normalized_rmse_tolerance = 0.3
+    ssim_index_tolerance = 0.6
+
+    populated_raw_data_ratio = get_populate_ratio(raw_data)
+    populated_downsized_data_ratio = get_populate_ratio(raw_data)
+    ratio_eval = are_numbers_within_tolerance(populated_raw_data_ratio, populated_downsized_data_ratio, ratio_tolerance)
+
+    filled_raw_data = raw_data.fillna(0)
+    filled_downsized_data = downsized_data.fillna(0)
+
+    # Resample raw_data to match the resolution of downsized_data
+    resampled_raw_data = filled_raw_data.interp_like(filled_downsized_data).fillna(0)
```

Comment:
So you're comparing the data resampled by GEE with the data resampled directly in xarray? This might have a few issues:

1) I think this still doesn't test that the resampling doesn't effect the overall indicator results (e.g. that the average albedo for an area is the roughly the same even after resampling). 
2) This call to `interp_like` is using just the default interpolation method (`linear`), but presumably we expect different datasets might require different interpolation methods, so we'd want to make sure those match.
3) I've noticed in the past xarray resampling just isn't the same as resampling used by geospatial libraries. It doesn't account for shifts in pixel coordinates in the same way. But I can't remember if I tried interp_like, so this might account for this.

My recommendation would be to compare statistics on the resampled layer (mean, min max, etc.) with the original resolution to make sure those aren't affected. But I'm glad to have a discussion if you think this is the best method.

Reply:
I had originally attempted to compare the resampled image to the original image, but ran into lots of issues due to the extent of dissimilarity with failures on several valid layers. But, I may have already solved the issues, so will attempt again.

Reply:
The earlier work was based on 4 quantiles, including Q50, so similar to your suggestion.

Reply:
@jterry64 I experimented with basic statistics like mean and quantiles and again reconfirmed that I could not find justifiable tolerances that worked with all layers using comparison of the "raw" and downsized images.

I instead expanded the comments and switched to quadratic interpolation. I think the evaluation is robust enough at this point to accept, but we can certainly improve the methodology over time.

Reply:
@kcartier-wri I guess that's to be expected, the cities team I think was being fairly liberal with resampling as needed for the indicators (usually because of computation timeouts with high resolution in GEE). I think that's fine, and maybe we can include a warning in the README or documentation that changing the spatial resolution can significantly change statistics, and to only use it for visualization?

Reply:
Also you knowledge of statistical tests seems much greater than mine, would love to discuss the methods you're using some time

Reply:
@jterry64 Happy to help. As I mentioned in Slack, I also experimented with normalized mean and stdev, but found that the range of layers and coarseness were challenging. In the end, I felt the code was not robust enough. 

Could you re-approve and resolve the conversations? Thanks.